### PR TITLE
Account for resends in queue state

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

File: src/streams/send.rs [hunk 1]
```diff
@@ -696,6 +703,28 @@ impl StreamTx {
 
         let mut snapshot = self.send_queue.snapshot(now);
 
+        // Outstanding resends
+        let mut resend_snapshot =
+            self.resends
+                .iter()
+                .fold(QueueSnapshot::default(), |mut snapshot, r| {
```
Comment (k0nserv) [inline src/streams/send.rs:710]: Doing this every time doesn't feel great for performance but neither does exposing `SendQueue::total` so we can call `increase` and `decrease` as the resends change.

We could also move `TotalQueue` out of `SendQueue` to `StreamTx`

File: src/streams/send.rs [hunk 2]
```diff
@@ -696,6 +703,28 @@ impl StreamTx {
...
+                        .first_unsent
+                        .map(|i| i.min(r.queued_at))
+                        .or(Some(r.queued_at));
+
+                    snapshot
+                });
+        resend_snapshot.created_at = now;
+        if resend_snapshot.packet_count > 0 {
+            resend_snapshot.priority = QueuePriority::Media;
+        }
```
Comment (algesten) [inline src/streams/send.rs:724]: Can you move this calculation to its own function? Just for tidyness.

**Review by algesten — APPROVED** [review]
LGTM

File: src/streams/send.rs (hunk 2 above)
Comment (k0nserv) [reply src/streams/send.rs:724]: Yep I was weighing that already

**Pull request merged**